zipkin-web/src/main/scala/com/twitter/zipkin/web/Handlers.scala

```diff
@@ -221,25 +215,13 @@ class Handlers(queryExtractor: QueryExtractor) {
         case None => EmptyTraces
       }
 
-      for (spans <- spansCall; traces <- tracesCall) yield {
-        val spanList = spans.toList map {
-          span => Map("name" -> span, "selected" -> (if (Some(span) == spanName) "selected" else ""))
-        }
-
+      for (traces <- tracesCall) yield {
+        val (annotations, binaryAnnotations) = queryExtractor.getAnnotations(req)
         var data = Map[String, Object](
-          ("serviceName" -> serviceName),
-          ("endTs" -> queryExtractor.getTimestampStr(req)),
-          ("annotationQuery" -> req.params.get("annotationQuery").getOrElse("")),
-          ("spans" -> spanList),
```

Comment:
Yes, the spans are loaded async from the frontend code. (I think they were loaded twice)